lib/admin/WP_Auth0_Admin_Features.php

```diff
@@ -55,131 +59,102 @@ public function init() {
+    );
   }
 
-  public function render_override_wp_avatars() {
-    $v = $this->options->get( 'override_wp_avatars' );
-
-    echo $this->render_a0_switch( "wpa0_override_wp_avatars", "override_wp_avatars", 1, !empty( $v ) );
-?>
-      <div class="subelement">
-        <span class="description"><?php echo __( 'Mark this if you want to override the WordPress avatar with the user\'s Auth0 profile avatar.', 'wp-auth0' ); ?></span>
-      </div>
-    <?php
+  public function render_override_wp_avatars( $args = array() ) {
+    $this->render_switch( $args[ 'label_for' ], $args[ 'opt_name' ] );
+    $this->render_field_description(
+      __( 'Overrides the WordPress avatar with the Auth0 profile avatar', 'wp-auth0' )
+    );
   }
 
-  public function render_fullcontact() {
-    $v = $this->options->get( 'fullcontact' );
-    $apikey = $this->options->get( 'fullcontact_apikey' );
-
-    echo $this->render_a0_switch( "wpa0_fullcontact", "fullcontact", 1, !empty( $v ) );
+  public function render_fullcontact( $args = array() ) {
+    $this->render_switch( $args['label_for'], $args['opt_name'], 'wpa0_fullcontact_key' );
+    $this->render_field_description(
+      __( 'Enriches your user profiles with the data provided by FullContact. ', 'wp-auth0' ) .
+      __( 'A valid FullContact API key is required for this to work. ', 'wp-auth0' ) .
+      __( 'For more details, see our ', 'wp-auth0' ) .
+      $this->get_docs_link(
+        'monitoring/track-signups-enrich-user-profile-generate-leads',
+        __( 'help page on tracking signups', 'wp-auth0' )
+      )
+    );
+  }
 
-?>
+  public function render_fullcontact_apikey( $args = array() ) {
+    $this->render_text_field( $args['label_for'], $args['opt_name'] );
+  }
 
-      <div class="subelement fullcontact <?php echo empty( $v ) ? 'hidden' : ''; ?>">
-        <label for="wpa0_fullcontact_key" id="wpa0_fullcontact_key_label">Enter your FullContact api key:</label>
-        <input type="text" id="wpa0_fullcontact_key" name="<?php echo $this->options->get_options_name(); ?>[fullcontact_apikey]" value="<?php echo $apikey; ?>" />
-      </div>
-
-      <div class="subelement">
-        <span class="description">
-          <?php echo __( 'Mark this if you want to enrich your users\' profiles with the data provided by FullContact. A valid api key is required. ', 'wp-auth0' ); ?>
-          <?php echo __( 'For more information, see our ', 'wp-auth0' ); ?>
-          <a target="_blank" href="https://auth0.com/docs/scenarios/mixpanel-fullcontact-salesforce#2-augment-user-profile-with-fullcontact-"><?php echo __( 'help page on FullContact integration with Auth0', 'wp-auth0' );?></a>
-        </span>
-      </div>
-    <?php
+  public function render_passwordless_enabled( $args = array() ) {
+    $this->render_switch( $args[ 'label_for' ], $args[ 'opt_name' ] );
+    $this->render_field_description(
+      __( 'Turn on Passwordless login (email or SMS) in the Auth0 form. ', 'wp-auth0' ) .
```

Comment:
What about the `MFA` grant? that needs to be enabled for the Application too, at least when calling this from an SDK. Test if creating a new Application leaves that grant enabled/disabled and if you can log in successfully using this wp-auth0

Reply:
No special grant needed for Passwordless, uses the Auth Code grant.